src/internal/base-component/__tests__/metrics.test.ts

```diff
@@ -332,6 +332,22 @@ describe('Client Metrics support', () => {
       });
     });
 
+    test('logs null prop values but not undefined', () => {
+      metrics.logComponentUsed('DummyComponentName', {
+        props: {},
+        metadata: { notDefined: undefined, nullValue: null },
+      });
+      checkMetric(`awsui_DummyComponentName_d10`, {
+        o: 'main',
+        s: 'DummyComponentName',
+        t: 'default',
+        a: 'used',
+        f: 'react',
+        v: '1.0',
+        c: { props: {}, metadata: { nullValue: null } },
```

Comment:
This test shows the use-case. Nulls end up in the serialized JSON, but undefined values do not.